src/vm/arm64/asmhelpers.asm

```diff
@@ -1326,5 +1329,62 @@ Fail
     LEAF_END
 #endif
 
+;
+; JIT Static access helpers when coreclr host specifies single appdomain flag 
+;
+
+; ------------------------------------------------------------------
+; void* JIT_GetSharedNonGCStaticBase(SIZE_T moduleDomainID, DWORD dwClassDomainID)
+
+    LEAF_ENTRY JIT_GetSharedNonGCStaticBase_SingleAppDomain
+    ; If class is not initialized, bail to C++ helper
+    add x2, x0, #DomainLocalModule__m_pDataBlob
+    ldrb w2, [x2, w1]
```

Comment:
why do you say that is invalid.
From section C6.6.87 in armv8 doc (not sure if you have same document)
LDRB <Wt>, [<Xn|SP>, <R><m>{, <extend> {<amount>}}]

<R> Is the index width specifier, encoded in the option field:
RESERVED when option = 00x
W when option = x10
X when option = x11
RESERVED when option = 10x
<m> Is the number [0-30] of the general-purpose index register or the name ZR (31), encoded in the Rm
field.